### PR TITLE
Make it more obvious when a community is locked

### LINKED ISSUE #2310: Visual display that a community is locked
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

There is currently no visual indicator that a post is locked, other than simply lacking the "Create a post" button. I think it would be good to have such a visual indicator, similiar to that of a post where it gets a padlock, to indicate that it is locked.

An example of a locked community that I am referring to is [!lemmyworld@lemmy.world](https://lemmy.world/c/lemmyworld).

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Closes #2310. This is what the community page looks like with my changes:
image

Note the red padlock next to the community name and the notice in the sidebar.

@K4LCIFER Do you think this makes it noticeable enough and looks alright?

### CODE REVIEW

**Comment by K4LCIFER** [comment]
I think that looks great! The padlock should also have a tooltip that says "Community locked", or somehting along those lines (if it doesn't already, of course).

File: src/shared/components/community/sidebar.tsx [hunk 1]
```diff
@@ -158,11 +159,24 @@ export class Sidebar extends Component<SidebarProps, SidebarState> {
...
+                  className="alert alert-info text-sm-start text-xs-center"
+                  role="alert"
+                >
+                  <Icon
+                    icon="lock"
+                    inline
+                    classes="me-sm-2 mx-auto d-sm-inline d-block"
+                  />
+                  <div className="d-inline">
+                    <strong className="fw-bold">Community locked:</strong> only
```
Comment (dessalines) [inline src/shared/components/community/sidebar.tsx:173]: Needs i18n.

**Comment by Nutomic** [comment]
Maybe it would also make sense to rename the setting from `posting_restricted_to_mods` to simply `locked`.

**Comment by dessalines** [comment]
I kinda like the verbose one, especially for code. Locked is somewhat unclear what that actually means, especially since its already used for posts. The front end can call that whatever it likes tho.

**Review by dessalines — APPROVED** [review]
👍

**Pull request merged**